src/java/htsjdk/tribble/index/IndexFactory.java

```diff
@@ -166,10 +166,11 @@ public static IndexType getIndexType(final BufferedInputStream is) {
      */
     public static Index loadIndex(final String indexFile) {
         final Index idx = null;
+        InputStream inputStream = null;
         BufferedInputStream bufferedInputStream = null;
         final LittleEndianInputStream dis = null;
         try {
-            InputStream  inputStream = ParsingUtils.openInputStream(indexFile);
+            inputStream = ParsingUtils.openInputStream(indexFile);
```

Comment:
There's still a leak here because the original code is rather odd.  It's opening an input stream here, and then possibly re-assigning `inputStream` immediately in the `if` below (I have no idea why).  Perhaps move line 173 into an `else` block after line 179?